pkg/proxy/userspace/roundrobin.go

```diff
@@ -120,6 +120,14 @@ func isSessionAffinity(affinity *affinityPolicy) bool {
 	return true
 }
 
+// ServiceHasEndpoints checks whether a service entry has endpoints.
+func (lb *LoadBalancerRR) ServiceHasEndpoints(svcPort proxy.ServicePortName) bool {
+	lb.lock.Lock()
+	defer lb.lock.Unlock()
+	state, exists := lb.services[svcPort]
+	return exists && state != nil && len(state.endpoints) > 0
```

Comment:
@DirectXMan12 can you refactor it or add a TODO item? Because, the condition will only get worse as other people will do the same and perform `nil` checks since the current code does it